### PR TITLE
[FEATURE]: DataProfilerStructuredDataAssistant Float Rule

### PR DESCRIPTION
**Pull request by micdavis**
Changes proposed in this pull request:

Added build float rule (in addition to numeric rule) to DataProfilerStructuredDataAssistant

This includes, a new rule in the capitalone_dataprofiler_greatexpectations section of contrib

Test coverage is complete and passing locally

- [x] My code follows the Great Expectations [style guide](https://docs.greatexpectations.io/docs/contributing/style_guides/code_style)

- [x] I have performed a [self-review](https://docs.greatexpectations.io/docs/contributing/contributing_checklist) of my own code

- [x] I have commented my code, particularly in hard-to-understand areas

- [x] I have added [unit tests](https://docs.greatexpectations.io/docs/contributing/contributing_test#writing-unit-and-integration-tests) where applicable and made sure that new and existing tests are passing.
After you submit your PR, keep the page open and **monitor the statuses of the various checks made by our continuous integration process at the bottom of the page. Please fix any issues that come up** and [reach out on Slack](https://greatexpectations.io/slack) if you need help. Thanks for contributing!

### CODE REVIEW

**Comment by micdavis** [comment]
This PR is a duplicate of PR #7731. I don't have access to `iniyam:data_assistant_float_rule_2`. So I pulled and I am making the needed changes on my fork.

File: contrib/capitalone_dataprofiler_expectations/capitalone_dataprofiler_expectations/rule_based_profiler/data_assistant/data_profiler_structured_data_assistant.py [hunk 1]
```diff
@@ -196,3 +198,114 @@ def _build_numeric_rule() -> Rule:
...
+            },
+        )
+
+        expect_column_stddev_to_be_between_expectation_configuration_builder: ExpectationConfigurationBuilder = DefaultExpectationConfigurationBuilder(
+            expectation_type="expect_column_stdev_to_be_between",
+            validation_parameter_builder_configs=validation_parameter_builder_configs,
+            column=f"{DOMAIN_KWARGS_PARAMETER_FULLY_QUALIFIED_NAME}{FULLY_QUALIFIED_PARAMETER_NAME_SEPARATOR_CHARACTER}column",
+            min_value=f"{data_profiler_profile_report_metric_single_batch_parameter_builder_for_validations.json_serialized_fully_qualified_parameter_name}{FULLY_QUALIFIED_PARAMETER_NAME_SEPARATOR_CHARACTER}{FULLY_QUALIFIED_PARAMETER_NAME_VALUE_KEY}.statistics.precision.std",
+            max_value=f"{data_profiler_profile_report_metric_single_batch_parameter_builder_for_validations.json_serialized_fully_qualified_parameter_name}{FULLY_QUALIFIED_PARAMETER_NAME_SEPARATOR_CHARACTER}{FULLY_QUALIFIED_PARAMETER_NAME_VALUE_KEY}.statistics.precision.std",
```
Comment (micdavis) [inline contrib/capitalone_dataprofiler_expectations/capitalone_dataprofiler_expectations/rule_based_profiler/data_assistant/data_profiler_structured_data_assistant.py:278]: @alexsherstinsky the different with the addition of the float rules is that we are checking a different metric in our report. For example at the end of line 277 and 278 we specify `statistics.precision.std` vs `statistics.stddev` which is a different metric in our report.

**Review by alexsherstinsky — APPROVED** [review]
LGTM!  Thank you very much, @micdavis !

**Comment by micdavis** [comment]
the "docs_integration" check seems to be stuck in "expected" status. Is there anything I need to do on my end?

**Review by alexsherstinsky — APPROVED** [review]
LGTM -- thank you very much!

**Pull request merged**